fix: Dispatch favorite event with an actual path

artonge opened the pull request:
The `$path` argument was added in https://github.com/nextcloud/server/pull/48612, but was never actually used by the callers. The path was therefore missing in the favorite/unfavorite events, which lead to a broken activity information.

I also added a fallback to handle `addToFavorites` and `removeFromFavorites`, which are part of a public API, and are calling `tagAs` and `untag` without `$path`.

Fix https://github.com/nextcloud/activity/issues/2134

artonge commented:
/backport to stable32

provokateurin commented on lib/public/ITags.php:
```diff
@@ -185,19 +185,23 @@ public function removeFromFavorites($objid);
 	 * @return boolean Returns false on database error.
+	 * @param string $path The optional path of the node. Used when dispatching the favorite change event.
 	 * @since 6.0.0
+	 * @since 31.0.0 Added the $path argument.
```
```suggestion
	 * @since 33.0.0 Added the $path argument.
```

provokateurin replied:
But the default value is changed in this PR

artonge replied:
True, will add a second line

provokateurin commented on lib/public/ITags.php:
```diff
@@ -185,19 +185,23 @@ public function removeFromFavorites($objid);
+	 * @param string $path The optional path of the node. Used when dispatching the favorite change event.
 	 * @return boolean
 	 * @since 6.0.0
+	 * @since 31.0.0 Added the $path argument.
```
```suggestion
	 * @since 33.0.0 Added the $path argument.
```

artonge replied:
It was added in 31 by https://github.com/nextcloud/server/pull/48612

artonge committed:
fix: Dispatch favorite event with an actual path

The `$path` argument was added in https://github.com/nextcloud/server/pull/48612, but was never actually used by the callers. The path was therefore missing in the favorite/unfavorite events, which lead to a broken activity information.

I also added a fallback to handle `addToFavorites` and `removeFromFavorites`, which are part of a public API, and are calling `tagAs` and `untag` without `$path`.

Fix https://github.com/nextcloud/activity/issues/2134

Signed-off-by: Louis Chemineau <louis@chmn.me>

AndyScherzinger reviewed (approved):
🐘
